CMake/FindJULIA.cmake

```diff
@@ -0,0 +1,45 @@
+# FindJULIA.cmake
+
+# Check if JULIA_HOME is set
```

Comment:
Please make sure the documentation in readthedocs will reflect these new options and requirements, i.e., having Julia. Also, include a link to help people get started/installing that so we do not have to repeat ourselves here.

Reply:
Please instruct how I can contribute to the 'readthedocs' , essentially,I need to know :
1. which is the root branch for the docs. is it still the 'docs' branch? In that case, how would we reconcile with the code changes with 'develop' to be the base of every feature branch. 
2. what is the structure of the docs directory? is there any written instruction on how to contribute to the docs directory? any restrictions and regulations we need to follow?

Additionally, I think we should probably look at `https://www.mkdocs.org`. With this, we don't have to translate markdown documents in to rst format, and the deployment of mkdocs can be done easily via GitHub Pages. Also, I would highly recommend maintaining a separate repo for the docs (including API docs, developers' guides, etc.) so that we can have clearer separation between source code management and document management.

Reply:
Let's have a zoom chat on this :-)

Reply:
Sure, the updated docs were merged to stable. So if you rebase, you should be getting the latest version.

I would strongly prefer we keep the docs within this project. We already overlook updating that in here. Having that shipped to another repo will not be beneficial. Also, lot so projects keep the docs within the project itself, and now that the readthedocs was integrated with the develop docs, we need access to the code to generate that, so it makes more sense to keep this here.

Reply:
I would strongly encourage a separation of the docs and the source code. 

I understand your point that we can keep track of doc changes and code changes from a single repo, however, in many cases, when unexpected conflicts/changes/reverts happens with the code but not the docs, the doc changes can be affected by the code changes, and vice versa. 

In addition, I  think when 'lot so projects keeps the docs within the project itself', it doesn't mean we necessarily have to follow such a practice. We have to choose the practice that works better for our team and we should not simply choose to follow a certain practice just because it is been practiced by some other projects. If just because of this, I also have seen a lot of other projects where docs are separated from code repos. But we have to understand and evaluate the pros and cons of a practice. 

Also, we cannot say since we need to generate 'readthedocs' on top of the docs directory of this repo, we have to keep the docs directory inside this repo. Even if we separate the docs to another repo, we can still generate the readthedocs. 


The drawbacks of having docs managed along with source code:
1. doc changes and source changes can happen at the same time in the same repo, and hence conflicts may happen between doc-dominant changes and code-dominant changes, and in this case, it would be very cumbersome to hold back either changes without being agile. 
2. The complexity of addressing conflicts between these two types of changes can really distract people from the quality of the docs, the code and the research work. This is not good for a growing project when more and more people joins in the project.  

The advantages of having separated repos for code and docs: 
1. Decoupling doc changes and code changes can absolutely help with better clarity of the team's workflow and can definitely reduce the complexity of any PR review. 
2. People can have better focus on the doc quality when reviewing doc changes, and have better focus on the code quality when reviewing code changes. 
3. To keep the document and the source code up-to-date, we have to make sure both changes have a smooth way to get in, instead of a lot of time being wasted in addressing conflicts. Otherwise, people will become frustrated in managing either of these, which is not something we would like to see.

Reply:
I don't understand how there are going to be conflicts between doc changes and source code changes, they are separate in different directories. As long as we only have one branch (develop) that pull requests go to, conflicts can be resolved easily.

Reply:
as long as these two directories are in the same repo, there can be chances two PRs with different changes on either directory may collide.

Reply:
We will merge PRs one by one, if two PRs have conflicts, it will get resolved when merging the second PR. 
Also, I don't see how having a separate docs repo helps, you need to merge 4 PRs instead of 2 and need to resolve the same conflicts.
We talked about PRs should be accompanied by corresponding document changes (when possible), having codes and docs in the same repo makes it easier to guarantee this.

Reply:
Reverting changes should happen very rarely if it happens at all. I don't think it is a good reason to separate code from docs.

Reply:
Anyway, suggestion is a suggestion. 

I have added related docs to reflect the changes about the Julia support this PR has. 
all PR comments are addressed. Please review and approve.

@jeanbez @houjun